### PR TITLE
DAG-1222: Add details of Recent Versions endpoint

### PR DESCRIPTION
**Pull request by dbradf**

### CODE REVIEW

File: tests/evergreen/test_version.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 from evergreen.manifest import Manifest
 from evergreen.metrics.versionmetrics import VersionMetrics
-from evergreen.version import Requester, Version
+from evergreen.version import RecentVersions, Requester, Version
```
Comment (trahman1318) [inline tests/evergreen/test_version.py:11]: We are testing `RecentVersions` which encompasses & indirectly tests `RecentVersionRow`. Would it be meaningful to add explicit tests for `RecentVersionRow`?

**Review by trahman1318 — APPROVED** [review]

File: tests/evergreen/test_version.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbradf) [reply tests/evergreen/test_version.py:11]: Given the simplicity of the code under test, I don't think we need explicit tests at every level of the code. If we can get good enough coverage at the level a user would interact with the code, then that tends to be ideal. If there were more complexity in `RecentVersionRow` that was difficult to test at a higher level, then I would agree.

File: tests/evergreen/test_version.py [hunk 2]
```diff
@@ -143,3 +143,23 @@ def test_get_metrics_completed(self, sample_version):
...
+
+
+class TestRecentVersions:
+    def test_row_map_should_be_querable(self, sample_recent_versions):
+        recent_versions = RecentVersions(sample_recent_versions, None)
+
+        assert len(recent_versions.row_map) == len(sample_recent_versions["rows"])
+        example_build_name = list(sample_recent_versions["rows"].keys())[2]
+        example_build_variant = sample_recent_versions["rows"][example_build_name]["build_variant"]
+        assert recent_versions.row_map[example_build_name].build_variant == example_build_variant
```
[the remark this answers is not in the document]
Comment (dbradf) [reply tests/evergreen/test_version.py:155]: We could. Given the simplicity of the code under test, I'm not sure it would be much more valuable.

**Comment by dbradf** [comment]
evergreen merge

**Pull request merged**